integration-tests/gradle/gradle6-test/single-platform/build.gradle.kts

```diff
@@ -22,36 +22,30 @@ plugins {
 }
 
 android {
-    compileSdk = Versions.Android.compileSdkVersion
+    compileSdk = 33
```

Comment:
We shouldn't necessarily use the versions of the SDK to compile these Gradle-version-specific projects. These projects tests Gradle-versions that specifically caused issues when collecting metrics or when enabling configuration cache. The can just as well use these hardcoded `compileSdk` value, etc. I have added a `integration-tests/gradle/current` that links in `buildSrc` and hence uses the exact versions as used to build the SDK. This doesn't guarantee anything, as the chances of a user actually running with these exact same versions is almost 0 🤷. But now we have a test of it ... and it will automatically follow along when bumping the constants in `Config.kt` or running `tools/update_gradle...`.

Reply:
If we really should do something efficient to guarantee coverage then we would need to define version spans for Gradle, Kotlin, AGP, compileSDK, Java-version, etc. and have parameterized execution of that. The issue is just that the project configuration and source code needs to change along with what ever version of the individual tool, and the amount of combinations would be insane, so 🤷

Reply:
Resolved offline. Leaving as is, makes the most sense 👍